### PR TITLE
fix(date-picker): reset datePicker when modelValue is undefined or an…

### LINKED ISSUE #1776: When in normal mode, not range or multiple, set undefined or empty string will not reset the date
# Bug report 🐞

## Actual results (or Errors)

image

## Reproduction link

[Reproduction link](https://varletjs.org/playground/#eyJBcHAudnVlIjoiPHNjcmlwdCBzZXR1cD5cbmltcG9ydCB7IG9uTW91bnRlZCB9IGZyb20gJ3Z1ZSc7XG5pbXBvcnQgeyByZWYgfSBmcm9tICd2dWUnXG5cbmNvbnN0IGRhdGUgPSByZWYoJzIwMjQtMDktMjMnKVxuXG5vbk1vdW50ZWQoKCkgPT4ge1xuICBzZXRUaW1lb3V0KCgpID0+IHtcbiAgICBkYXRlLnZhbHVlID0gIHVuZGVmaW5lZFxuICB9LCAzMDAwKVxufSlcbjwvc2NyaXB0PlxuXG48dGVtcGxhdGU+XG4gIDx2YXItZGF0ZS1waWNrZXIgdi1tb2RlbD1cImRhdGVcIiBlbGV2YXRpb24gLz5cbjwvdGVtcGxhdGU+XG4iLCJBcHBXcmFwcGVyLnZ1ZSI6IjxzY3JpcHQgc2V0dXA+XG5pbXBvcnQgQXBwIGZyb20gJy4vQXBwLnZ1ZSdcbmltcG9ydCB7IGluc3RhbGxWYXJsZXRVSSB9IGZyb20gJy4vdmFybGV0LXJlcGwtcGx1Z2luLmpzJ1xuXG5pbnN0YWxsVmFybGV0VUkoKVxuPC9zY3JpcHQ+XG5cbjx0ZW1wbGF0ZT5cbiAgPEFwcCAvPlxuPC90ZW1wbGF0ZT5cbiIsInZhcmxldC1yZXBsLXBsdWdpbi5qcyI6ImltcG9ydCBWYXJsZXRVSSBmcm9tICdAdmFybGV0L3VpJ1xuaW1wb3J0IHsgZ2V0Q3VycmVudEluc3RhbmNlIH0gZnJvbSAndnVlJ1xuXG5jb25zdCB2YXJsZXRDc3MgPSAnaHR0cHM6Ly9jZG4uanNkZWxpdnIubmV0L25wbS9AdmFybGV0L3VpL2VzL3N0eWxlLmNzcydcbmNvbnN0IHZhcmxldFRvdWNoRW11bGF0b3IgPSAnaHR0cHM6Ly9jZG4uanNkZWxpdnIubmV0L25wbS9AdmFybGV0L3RvdWNoLWVtdWxhdG9yL2lpZmUuanMnXG5cbmF3YWl0IGFwcGVuZFN0eWxlKClcblxuZXhwb3J0IGZ1bmN0aW9uIGluc3RhbGxWYXJsZXRVSSgpIHtcbiAgY29uc3QgeyBwYXJlbnQgfSA9IHdpbmRvd1xuXG4gIGNvbnN0IHN0eWxlID0gZG9jdW1lbnQuY3JlYXRlRWxlbWVudCgnc3R5bGUnKVxuICBzdHlsZS5pbm5lckhUTUwgPSBgXG4gICAgYm9keSB7XG4gICAgICBtaW4taGVpZ2h0OiAxMDB2aDtcbiAgICAgIHBhZGRpbmc6IDE2cHg7XG4gICAgICBtYXJnaW46IDA7XG4gICAgICBjb2xvcjogdmFyKC0tY29sb3ItdGV4dCk7XG4gICAgICBiYWNrZ3JvdW5kLWNvbG9yOiB2YXIoLS1jb2xvci1ib2R5KTtcbiAgICB9XG5cbiAgICAqOjotd2Via2l0LXNjcm9sbGJhciB7XG4gICAgICBkaXNwbGF5OiBub25lO1xuICAgIH1cbiAgYFxuICBkb2N1bWVudC5oZWFkLmFwcGVuZENoaWxkKHN0eWxlKVxuXG4gIGNvbnN0IHNjcmlwdCA9IGRvY3VtZW50LmNyZWF0ZUVsZW1lbnQoJ3NjcmlwdCcpXG4gIHNjcmlwdC5zcmMgPSB2YXJsZXRUb3VjaEVtdWxhdG9yXG4gIGRvY3VtZW50LmJvZHkuYXBwZW5kQ2hpbGQoc2NyaXB0KVxuXG4gIGNvbnN0IHRoZW1lTWFwID0ge1xuICAgIGxpZ2h0VGhlbWU6IG51bGwsXG4gICAgZGFya1RoZW1lOiBWYXJsZXRVSS5UaGVtZXMuZGFyayxcbiAgICBtZDNMaWdodFRoZW1lOiBWYXJsZXRVSS5UaGVtZXMubWQzTGlnaHQsXG4gICAgbWQzRGFya1RoZW1lOiBWYXJsZXRVSS5UaGVtZXMubWQzRGFyayxcbiAgfVxuXG4gIGNvbnN0IHRoZW1lID0gcGFyZW50LmRvY3VtZW50LmRvY3VtZW50RWxlbWVudC5nZXRBdHRyaWJ1dGUoJ3RoZW1lJylcblxuICBpZiAodGhlbWUpIHtcbiAgICBWYXJsZXRVSS5TdHlsZVByb3ZpZGVyKHRoZW1lTWFwW3RoZW1lXSlcbiAgfVxuXG4gIHdpbmRvdy5hZGRFdmVudExpc3RlbmVyKCdtZXNzYWdlJywgKHsgZGF0YSB9KSA9PiB7XG4gICAgaWYgKGRhdGEuYWN0aW9uID09PSAndGhlbWUtY2hhbmdlJykge1xuICAgICAgVmFybGV0VUkuU3R5bGVQcm92aWRlcih0aGVtZU1hcFtkYXRhLnZhbHVlXSlcbiAgICB9XG4gIH0pXG5cbiAgZG9jdW1lbnQuYWRkRXZlbnRMaXN0ZW5lcignY2xpY2snLCAoKSA9PiB7XG4gICAgd2luZG93LnBhcmVudC5kb2N1bWVudC5kaXNwYXRjaEV2ZW50KG5ldyBFdmVudCgnY2xpY2snKSlcbiAgfSlcblxuICBjb25zdCBpbnN0YW5jZSA9IGdldEN1cnJlbnRJbnN0YW5jZSgpXG4gIGluc3RhbmNlLmFwcENvbnRleHQuYXBwLnVzZShWYXJsZXRVSSlcbn1cblxuZXhwb3J0IGZ1bmN0aW9uIGFwcGVuZFN0eWxlKCkge1xuICByZXR1cm4gbmV3IFByb21pc2UoKHJlc29sdmUsIHJlamVjdCkgPT4ge1xuICAgIGNvbnN0IGxpbmsgPSBkb2N1bWVudC5jcmVhdGVFbGVtZW50KCdsaW5rJylcbiAgICBsaW5rLnJlbCA9ICdzdHlsZXNoZWV0J1xuICAgIGxpbmsuaHJlZiA9IHZhcmxldENzc1xuICAgIGxpbmsub25sb2FkID0gcmVzb2x2ZVxuICAgIGxpbmsub25lcnJvciA9IHJlamVjdFxuICAgIGRvY3VtZW50LmhlYWQuYXBwZW5kQ2hpbGQobGluaylcbiAgfSlcbn0iLCJpbXBvcnQtbWFwLmpzb24iOiJ7XG4gIFwiaW1wb3J0c1wiOiB7XG4gICAgXCJAdmFybGV0L3VpXCI6IFwiaHR0cHM6Ly9jZG4uanNkZWxpdnIubmV0L25wbS9AdmFybGV0L3VpL2VzL3ZhcmxldC5lc20uanNcIixcbiAgICBcIkB2YXJsZXQvdWkvanNvbi9hcmVhLmpzb25cIjogXCIuL3ZhcmxldC1hcmVhLmpzXCIsXG4gICAgXCJ6b2RcIjogXCJodHRwczovL2Nkbi5qc2RlbGl2ci5uZXQvbnBtL3pvZC9saWIvaW5kZXgubWpzXCJcbiAgfVxufSIsInRzY29uZmlnLmpzb24iOiJ7XG4gIFwiY29tcGlsZXJPcHRpb25zXCI6IHtcbiAgICBcImFsbG93SnNcIjogdHJ1ZSxcbiAgICBcImNoZWNrSnNcIjogdHJ1ZSxcbiAgICBcImpzeFwiOiBcIlByZXNlcnZlXCIsXG4gICAgXCJ0YXJnZXRcIjogXCJFU05leHRcIixcbiAgICBcIm1vZHVsZVwiOiBcIkVTTmV4dFwiLFxuICAgIFwibW9kdWxlUmVzb2x1dGlvblwiOiBcIkJ1bmRsZXJcIixcbiAgICBcImFsbG93SW1wb3J0aW5nVHNFeHRlbnNpb25zXCI6IHRydWVcbiAgfSxcbiAgXCJ2dWVDb21waWxlck9wdGlvbnNcIjoge1xuICAgIFwidGFyZ2V0XCI6IDMuM1xuICB9XG59In0=)

### PR DESCRIPTION
**Pull request by windowdotonload**
… empty string

## Checklist

List of tasks you have already done and plan to do.

- [ ] Fix linting errors
- [ ] Tests have been added / updated (or snapshots)

## Change information

Describe your modifications here.

## Issues

close #1776

## Related Links

Links related to this pr.

### CODE REVIEW

**Comment by BeADre** [comment]
Please fix the test case error

**Comment by windowdotonload** [comment]
> Please fix the test case error

fix done

**Review by BeADre — APPROVED** [review]

**Comment by BeADre** [comment]
cc @chouchouji

File: packages/varlet-ui/src/date-picker/DatePicker.vue [hunk 1]
```diff
@@ -513,8 +513,9 @@ export default defineComponent({
       rangeDate.value = formatDateList.filter((date) => date !== 'Invalid Date')
     }
 
-    function dateInit(value: string) {
-      const formatDate = dayjs(value).format('YYYY-MM-D')
+    function dateInit(value: string | undefined) {
+      const handleValue = value ? dayjs(value) : dayjs()
+      const formatDate = dayjs(handleValue).format('YYYY-MM-D')
```
Comment (chouchouji) [inline packages/varlet-ui/src/date-picker/DatePicker.vue:518]: If the value is undefined or an empty string, `handleValue` will be `dayjs()`. This code `dayjs(dayjs())`  seems not necessary.

It looks like no differences.
image

File: packages/varlet-ui/src/date-picker/__tests__/index.spec.js [hunk 2]
```diff
@@ -422,3 +422,27 @@ test('test datePicker titleColor', async () => {
...
+      return {
+        date: '2024-09-23',
+      }
+    },
+    template: `<var-date-picker v-model="date" />`,
+  })
+
+  expect(wrapper.html()).toMatchSnapshot()
+
+  await wrapper.setData({ time: undefined })
```
Comment (chouchouji) [inline packages/varlet-ui/src/date-picker/__tests__/index.spec.js:441]: The data you set is `date`, not `time`.  Adding snapshot after `setData` may be a good practice to check whether the time is right.

**Comment by windowdotonload** [comment]
thanks for your review , I've changed sth. @chouchouji

**Review by chouchouji — APPROVED** [review]

**Comment by chouchouji** [comment]
> thanks for your review , I've changed sth. @chouchouji

Thanks for your great work!

**Pull request merged**